src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/EmbeddedLanguages/VirtualChars/CSharpVirtualCharService.cs

```diff
@@ -145,6 +161,72 @@ private static VirtualCharSequence TryConvertRawStringToVirtualChars(SyntaxToken
             return CreateVirtualCharSequence(tokenText, offset, startIndexInclusive, endIndexExclusive, result);
         }
 
+        private static VirtualCharSequence TryConvertMultiLineRawStringToVirtualChars(SyntaxToken token)
+        {
+            // The containing expression for this token.  This is needed so that we can determine the indentation
+            // whitespace based on the last line of the containing multiline literal.
+            ExpressionSyntax parentExpression = token.Kind() is SyntaxKind.MultiLineRawStringLiteralToken or SyntaxKind.UTF8MultiLineRawStringLiteralToken
+                ? (LiteralExpressionSyntax)token.GetRequiredParent()
+                : (InterpolatedStringExpressionSyntax)token.GetRequiredParent().GetRequiredParent();
+
+            // Whether or not the token contains the delimiters, or if the parent owns them.  This will affect if we
+            // need to skip looking at the start/end line of the token.
+            var hasDelimiters = parentExpression is LiteralExpressionSyntax;
+
+            // if this is the first text content chunk of the multi-line literal.  The first chunk contains the leading
+            // indentation of the line it's on (which thus must be trimmed), while all subsequent chunks do not (because
```

Comment:
I am confused by this method, but it may just be me...
If you have
```
var xml = $$"""
          <element attr="{{_elementName}}" id="someId">
            <body>
            </body>
          </element>
          """;
```

I assume the first chunk is
```
         <element attr="
```

and the second is 
```
 id="someId">
```
So it makes sense that you don't want to remove indentation in the second bit.
But in the loop below you're going line by line anyway to remove indentation, so how does the isFirstChunk matter?  Since they're on the same line how would the indentation ever be removed for the second chunk?

Or is the second chunk a separate call to this method?  Hence when you enter this for the second chunk, you don't have the text before the second chunk, but its on its own line and so without this check the indentation in it gets removed?  Then my question is couldn't you have multiple first chunks on each line?

Reply:
> and the second is

teh second is not that.  The second would be:

```
 id="someId">
            <body>
            </body>
          </element>
```